api/controllers/FeedbackController.ts

```diff
@@ -21,9 +23,10 @@ export class FeedbackController {
   }
 
   @Get()
-  async getFeedback(@AuthenticatedUser() user: UserModel): Promise<GetFeedbackResponse> {
+  async getFeedback(@QueryParams() options: FeedbackSearchOptions,
+    @AuthenticatedUser() user: UserModel): Promise<GetFeedbackResponse> {
     const canSeeAllFeedback = PermissionsService.canRespondToFeedback(user);
```

Comment:
I feel as though this is a bit inconsistent (technically, a standard user would result in the `canSeeAllFeedback` variable being false. However, they can still view their own feedback, just not the feedback of others. `canRespondToFeedback` is a bit too generalizing for readability in my opinion, but I'm curious what you think makes sense here.

Also, a potential bug/inconsistency. As a standard user, using query params to see my own feedback doesn't work since it just default queries all of the user's feedback (I think?) :

<img width="853" alt="image" src="https://github.com/acmucsd/membership-portal/assets/16010723/96dd13aa-2ad6-4246-86c3-8bdb677f6ef0">

Reply:
It works fine on my end--maybe you didn't change the access token? For me, upon running the API, hitting the GET route as a standard user returns empty json, and all after submitting a few feedbacks, all of the user's feedbacks are returned properly regardless of the queryparams. 
<img width="896" alt="Screenshot 2024-03-28 at 1 22 13 AM" src="https://github.com/acmucsd/membership-portal/assets/69327109/5b2d7103-4411-4df9-a534-25bec118c669">
<img width="906" alt="Screenshot 2024-03-28 at 1 23 06 AM" src="https://github.com/acmucsd/membership-portal/assets/69327109/440c5602-3b4e-4ae2-9337-a0d8ab28c573">

Reply:
I agree with canRespondToFeedback not being the most descriptive variable name, especially since the method itself simply checks if the user is an admin. This was in the previous version of the function and I didn't change it. I've refactored it to canQueryAllFeedbacks, let me know if that's better or if I should rename it to something else.